File: apps/api/src/services/signup.ts
import { db } from '@/lib/db';

export const signup = async (req, res) => {
  const { email, providerId } = req.body;

  const user = await db
    .insertInto('users')
    .values({
      email,
      username: email.split('@')[0],
      provider_id: providerId,
    })
    .onConflict((oc) => 
      oc.column('email').doUpdateSet({
        provider_id: providerId,
      })
    )
    .returning(['id'])
    .executeTakeFirst();

  return res.json(user);
}

File: apps/backend/backend/services/signup.py
from backend.libs.db import db
from flask import request

def signup():
  email = request.json.get('email')
  provider_id = request.json.get('providerId')
  username = email.split('@')[0]

  user = db.insert('users').values(email=email, username=username, provider_id=provider_id).on_conflict_do_update(target=['email'], set_={'provider_id': provider_id}).returning('id').execute_one()

  return user

